### PR TITLE
fix click_deploy_test

### PR DESCRIPTION
**Pull request by lluunn**
- Change the srcDir and set GOPATH in click_deploy_test.jsonnet so that we can build with go modules
- update build_image.sh
- test_deploy_app.py also does cleanup first in e2e mode

For https://github.com/kubeflow/kubeflow/issues/2364
For https://github.com/kubeflow/testing/issues/301

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2400)

### CODE REVIEW

**Comment by lluunn** [comment]
/assign @kunmingg 
/assign @jlewi 
ready for review

File: testing/test_deploy_app.py [hunk 1]
```diff
@@ -472,12 +480,12 @@ def main(unparsed_args=None):
...
   parser.add_argument(
     "--project",
-    default="kubeflow-ci-deploy",
+    default="kubeflow-ci-deployment",
     type=str,
     help="e2e test project id")
   parser.add_argument(
     "--project_number",
-    default="453914067825",
+    default="29647740582",
```
[the remark this answers is not in the document]
Comment (lluunn) [reply testing/test_deploy_app.py]: It's the same. project id and number are both passed as request. @kunmingg is it needed?

File: testing/workflows/components/click_deploy_test.jsonnet [hunk 2]
```diff
@@ -25,7 +25,7 @@ local artifactsDir = outputDir + "/artifacts";
 // Source directory where all repos should be checked out
 local srcRootDir = testDir + "/src";
 // The directory containing the kubeflow/kubeflow repo
-local srcDir = srcRootDir + "/kubeflow/kubeflow";
+local srcDir = srcRootDir + "/github.com/kubeflow/kubeflow";
```
[the remark this answers is not in the document]
Comment (lluunn) [reply testing/workflows/components/click_deploy_test.jsonnet]: Yes, it's for gopath

[the remark this answers is not in the document]
Comment (lluunn) [reply testing/workflows/components/click_deploy_test.jsonnet]: updated

File: testing/test_deploy_app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kunmingg) [reply testing/test_deploy_app.py]: Yes, currently we pass both project id and number to deploy API, which can be simplified.

**Comment by lluunn** [comment]
@kunmingg yes , I was running it in new project

**Comment by kunmingg** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**